### PR TITLE
🌱 Add Proxmox provider quickstart

### PR DESCRIPTION
**Pull request by mcbenjemaa**
**What this PR does / why we need it**:

This PR adds:
* quickstat documentation for the Proxmox provider: https://github.com/ionos-cloud/cluster-api-provider-proxmox

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area docs

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -740,6 +740,38 @@ clusterctl init --infrastructure vsphere
...
+{{#tab Proxmox}}
+
+First, we need to add the Proxmox infrastructure provider and the IPAM provider to your clusterctl config file `~/.cluster-api/clusterctl.yaml`:
+```yaml
+providers:
+  - name: in-cluster
+    url: https://github.com/kubernetes-sigs/cluster-api-ipam-provider-in-cluster/releases/latest/ipam-components.yaml
+    type: IPAMProvider
+  - name: proxmox
+    url: https://github.com/ionos-cloud/cluster-api-provider-proxmox/releases/latest/infrastructure-components.yaml
```
Comment (sbueringer) [inline docs/book/src/user/quick-start.md:753]: You might want to add the Proxmox provider to clusterctl so you don't need this step.

But as long we're waiting for the IPAM provider to be added it doesn't significantly simplify these steps anyway :)

**Review by sbueringer — COMMENTED** [review]
Nice, thx!

File: docs/book/src/user/quick-start.md (hunk 1 above)
Comment (mcbenjemaa) [reply docs/book/src/user/quick-start.md:753]: yeah, I'm looking to add it in clusterctl, but waiting for IPAM as well.

Comment (sbueringer) [reply docs/book/src/user/quick-start.md:753]: Feel free to add in independently of the IPAM provider to clusterctl if you want

Comment (mcbenjemaa) [reply docs/book/src/user/quick-start.md:753]: so, you mean I can add the Proxmox in defaults?
https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/client/config/providers_client.go

and only add ipam into clusterctl config

Comment (sbueringer) [reply docs/book/src/user/quick-start.md:753]: Exactly

Comment (mcbenjemaa) [reply docs/book/src/user/quick-start.md]: done

**Pull request merged**